chain/chunks/src/lib.rs

```diff
@@ -730,34 +732,28 @@ impl ShardsManager {
                     let epoch_id = unwrap_or_return!(
                         runtime_adapter.get_epoch_id_from_prev_block(known_header.prev_hash())
                     );
-                    let block_producer =
-                        unwrap_or_return!(runtime_adapter.get_block_producer(&epoch_id, height));
```

Comment:
@bowenwang1996 here is where the bug regarding chunk producer signature was, I believe this line should be getting chunk producer instead of block producer.

The test on this function didn't catch this because the test only had one validators. I'll change that.

Reply:
Interesting. I wonder why we never caught this in any tests

Reply:
I think because
1) block producer and chunk producers are the same because of how we assign block producer and chunk producers. 
2) there is not much test for testing the code regarding store_partial_encoded_chunks itself. And even if this part of logic is not exactly correct, it won't affect block processing that much because the partial chunks can be requested again when the block is ready. This will only affect block processing delay, and not many tests in our code check that.

I know I'm not adding any test here for this logic right now. Because
1) As I said, even if the logic is not entirely correct, it won't affect block processing correctness
2) I'm thinking getting rid of this part of logic of `store_partial_encoded_chunks` logic entirely, because they could be just processed instead of stored. I want to make that change after I finish https://github.com/near/nearcore/pull/5224
3) I'll add tests in https://github.com/near/nearcore/pull/5224 to make sure for orphans, chunks are requested and stored so they don't need to be requested again when the orphan is ready to be processed.